traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateRegistry.java

```diff
@@ -103,7 +105,7 @@ synchronized public void importCertificateDataList(final List<CertificateData> c
 		previousData.clear();
 		for (final CertificateData certificateData : certificateDataList) {
 			final String alias = certificateData.alias();
-			if (!previousData.containsKey(alias)) {
+			if (!previousData.containsKey(alias) && master.containsKey(alias)) {
```

Comment:
Since I'm not all that familiar with this part of TR, why is this change necessary? What is this code doing with `master` and `previousData`?

Reply:
`PreviousData` is the current map of `CertificateData` which is used to compare with the input list (`certificateDataList`) to find differences which need to be updated or deleted. `master` is the subset of valid `HandshakeData` which were successfully converted from the `CertificateData` in the input `certificateDataList`. The added condition in the `if` statement is necessary because of the new certificate validation feature could have filtered some of the entries out of the input `certificateDataList'.

Reply:
Cool, makes sense. Thanks for helping me learn!